src/group/managed_group/callbacks.rs

```diff
@@ -22,33 +50,75 @@ pub struct ManagedGroupCallbacks {
-        validate_remove: Option<ValidateRemove>,
-        member_added: Option<MemberAdded>,
-        member_removed: Option<MemberRemoved>,
-        member_updated: Option<MemberUpdated>,
-        app_message_received: Option<AppMessageReceived>,
-        invalid_message_received: Option<InvalidMessageReceived>,
-        error_occured: Option<ErrorOccured>,
-    ) -> Self {
+impl<'a> ManagedGroupCallbacks {
+    pub fn new() -> Self {
         Self {
-            validate_add,
-            validate_remove,
-            member_added,
-            member_removed,
-            member_updated,
-            app_message_received,
-            invalid_message_received,
-            error_occured,
+            validate_add: None,
+            validate_remove: None,
+            member_added: None,
+            member_removed: None,
+            member_updated: None,
+            app_message_received: None,
+            invalid_message_received: None,
+            error_occured: None,
         }
     }
+    /// Validator function for AddProposals
+    pub fn with_validate_add(mut self, validate_add: ValidateAdd) -> Self {
+        self.validate_add = Some(validate_add);
+        self
+    }
+    /// Validator function for RemoveProposals
+    pub fn with_validate_remove(mut self, validate_remove: ValidateRemove) -> Self {
+        self.validate_remove = Some(validate_remove);
+        self
+    }
+    /// Event listener function for AddProposals
+    pub fn with_member_added(mut self, member_added: MemberAdded) -> Self {
+        self.member_added = Some(member_added);
+        self
+    }
+    /// Event listener function for RemoveProposals when a member was removed
+    pub fn with_member_removed(mut self, member_removed: MemberRemoved) -> Self {
+        self.member_removed = Some(member_removed);
+        self
+    }
+    /// Event listener function for UpdateProposals
+    pub fn with_member_updated(mut self, member_updated: MemberUpdated) -> Self {
+        self.member_updated = Some(member_updated);
+        self
+    }
+    /// Event listener function for application messages
+    pub fn with_app_message_received(mut self, app_message_received: AppMessageReceived) -> Self {
+        self.app_message_received = Some(app_message_received);
+        self
+    }
+    /// Event listener function for invalid messages
+    pub fn with_invalid_message_received(
+        mut self,
+        invalid_message_received: InvalidMessageReceived,
+    ) -> Self {
+        self.invalid_message_received = Some(invalid_message_received);
+        self
+    }
+    /// Event listener function for errors that occur
+    pub fn with_error_occured(mut self, error_occured: ErrorOccured) -> Self {
+        self.error_occured = Some(error_occured);
+        self
+    }
 }
 
 /// This enum lists the 4 different variants of a removal, depending on who the
 /// remover and who the leaver is.
+///
+///  - `WeLeft`: We previously issued a RemoveProposal for ourselves and this
```

Comment:
These doc comments should go to the individual elements of the enum to get the documentation displayed correctly.